.github/workflows/tests.yaml

```diff
@@ -44,8 +38,59 @@ jobs:
       - name: Start Weaviate
         run: ./test/start_containers.sh
       - name: Run tests
-        run: |
-          go test -v ./weaviate/...
-          ( for pkg in $(go list ./... | grep 'weaviate-go-client/v4/test'); do if ! go test -v -count 1 -race "$pkg"; then echo "Test for $pkg failed" >&2; false; exit; fi; done)
+        run: go test -v -count 1 -race test/auth/*.go
       - name: Stop Weaviate
         run: ./test/stop_containers.sh
+  tests:
+    name: Integration
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        auth_integration: [ "auth_enabled", "auth_disabled" ]
+    env:
+      EXTERNAL_WEAVIATE_RUNNING: false
+      OKTA_DUMMY_CI_PW: ${{ secrets.OKTA_DUMMY_CI_PW }}
+      INTEGRATION_TESTS_AUTH: ${{ matrix.auth_integration }}
+      OPENAI_APIKEY: ${{ secrets.OPENAI_APIKEY }}
+    steps:
+      - uses: actions/checkout@v3
+      - name: Login to Docker Hub
+        uses: docker/login-action@v2
+        if: ${{ !github.event.pull_request.head.repo.fork }}
+        with:
+          username: ${{secrets.DOCKER_USERNAME}}
+          password: ${{secrets.DOCKER_PASSWORD}}
+      - name: Set up Go
+        uses: actions/setup-go@v3
+        with:
+          go-version: 1.19
+          cache: true
+      - name: Run tests
+        run: ( for pkg in $(go list ./... | grep 'weaviate-go-client/v4/test/' | grep -v 'auth'); do if ! go test -v -count 1 -race "$pkg"; then echo "Test for $pkg failed" >&2; false; exit; fi; done)
+  unit-tests:
+    name: Unit tests
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Set up Go
+        uses: actions/setup-go@v3
+        with:
+          go-version: 1.19
+          cache: true
+      - name: Run tests
+        run: go test -v ./weaviate/...
+  tests-deprecated:
+    runs-on: ubuntu-latest
+    env:
+      EXTERNAL_WEAVIATE_RUNNING: false
+    steps:
+      - uses: actions/checkout@v3
+      - name: Login to Docker Hub
+        uses: docker/login-action@v2
+        if: ${{ !github.event.pull_request.head.repo.fork }}
+        with:
+          username: ${{secrets.DOCKER_USERNAME}}
+          password: ${{secrets.DOCKER_PASSWORD}}
+      - name: Run tests
+        run:
+          go test -v -count 1 -race test_deprecated/*.go
```

Comment:
Actually if we would have `./tools/run_tests.sh` script then we could add modifiers to them for all tests, so that for example this part would be invoked with `./tools/run_tests.sh --deprecated`

Reply:
done, please have a look :)